### PR TITLE
Run the Housing Finance API locally

### PR DESCRIPTION
**Pull request by adamtry**
### Simplify running the HFS API via Docker
- Fixed the file paths and ports in the docker compose
- Fixed the file paths and ports in the Dockerfile
- Introduced `finance_database.mk` to allow port forwarding to a deployed database

### Simplify running the HFS API directly with `dotnet run`
- Added the `dotenv.net` Nuget package to load a `.env` file in the root directory with the required environment variables
- Added a `.env.sample` file to indicate the required variables to set

### General
- Made it easier to generate the `CONNECTION_STRING` environment variable for a given profile with a Make command
- Updated the README to give a step by step walkthrough of how to run the API locally in both ways

### Improve development process
- Engineers should be able to manually test the system as a whole, especially after making changes, without needing to deploy to AWS
- Engineers should be able to run the API in debug mode using IDEs

[POH-155]: https://hackney.atlassian.net/browse/POH-155?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,15 +168,18 @@ If changes to the database schema are made then the docker image for the databas
...
 
 ### Other Contacts
 
-- **Rashmi Shetty**, Product Owner at London Borough of Hackney (rashmi.shetty@hackney.gov.uk)
+- **Selwyn Preston**, Head of Engineering at London Borough of Hackney (selwyn.preston@hackney.gov.uk)
 
 [docker-download]: https://www.docker.com/products/docker-desktop
+[docker-compose-download]: https://docs.docker.com/compose/install/
 [universal-housing-simulator]: https://github.com/LBHackney-IT/lbh-universal-housing-simulator
 [made-tech]: https://madetech.com/
```
Comment (humulla) [inline README.md:182]: Do we need made tech here?

Comment (adamtry) [reply README.md]: Removed the link in https://github.com/LBHackney-IT/housing-finance-interim-api/pull/114/commits/945916eac09932b12469c4a16f5f23fc5a40090d

**Review by humulla — COMMENTED** [review]
This PR should be to development branch first

**Review by FaisalHackney — APPROVED** [review]

File: HousingFinanceInterimApi/Startup.cs [hunk 2]
```diff
@@ -44,6 +43,9 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            // Load environment variables from .env file in parent directory if it exists
+            DotEnv.Fluent().WithEnvFiles(Path.Combine(Directory.GetCurrentDirectory(), "../.env")).Load();
```
Comment (Duslerke) [inline HousingFinanceInterimApi/Startup.cs:47]: Maybe add 'if environment = local'?

File: README.md [hunk 3]
```diff
@@ -114,15 +180,17 @@ If changes to the database schema are made then the docker image for the databas
 
 ### Active Maintainers
 
-- **Selwyn Preston**, Lead Developer at London Borough of Hackney (selwyn.preston@hackney.gov.uk)
-- **Mirela Georgieva**, Lead Developer at London Borough of Hackney (mirela.georgieva@hackney.gov.uk)
-- **Matt Keyworth**, Lead Developer at London Borough of Hackney (matthew.keyworth@hackney.gov.uk)
+- **Faisal Gazi**, Lead Software Engineer at London Borough of Hackney (faisal.gazi@hackney.gov.uk)
+- **Humairaa Mulla**, Senior Software Engineer at London Borough of Hackney (humairaa.mulla@hackney.gov.uk)
+- **George Schena**, Senior Software Engineer at London Borough of Hackney (george.schena@hackney.gov.uk)
```
Comment (Duslerke) [inline README.md:185]: George is a maintainer & I'm not? :sweat:

File: docker-compose.yml [hunk 4]
```diff
@@ -4,20 +4,32 @@ services:
...
+
+#    Dev database not working
+#    links:
+#      - dev-database
+
   dev-database:
     env_file:
       - database.env
-    image: microsoft/mssql-server-windows-express
+    image: mcr.microsoft.com/mssql/server:2019-latest
```
Comment (Duslerke) [inline docker-compose.yml:32]: what's the difference between the two images?

File: docker-compose.yml [hunk 5]
```diff
@@ -4,20 +4,32 @@ services:
...
+      - GOOGLE_API_KEY=${GOOGLE_API_KEY}
+      - WAIT_DURATION=${WAIT_DURATION}
+      - BATCH_SIZE=${BATCH_SIZE}
+      - CASH_FILE_REGEX=${CASH_FILE_REGEX}
+      - HOUSING_FILE_REGEX=${HOUSING_FILE_REGEX}
+      - ACCEPTED_ORIGINS=${ACCEPTED_ORIGINS}
+      - CHARGES_BATCH_YEARS=${CHARGES_BATCH_YEARS}
+      - S3_BUCKET_NAME=${S3_BUCKET_NAME}
+      - S3_OBJECT_PREFIX=${S3_OBJECT_PREFIX}
+      - HOUSING_FINANCE_ALLOWED_GROUPS=${HOUSING_FINANCE_ALLOWED_GROUPS}
```
Comment (Duslerke) [inline docker-compose.yml:23]: If we're going this route, the they do we need `dotenv` package?

**Comment by Duslerke** [comment]
Wait, this branch is tainted.

**Pull request closed without merging**